sklearn/linear_model/cd_fast.pyx

```diff
@@ -120,6 +129,92 @@ cdef extern from "cblas.h":
     void dscal "cblas_dscal"(int N, double alpha, double *X, int incX) nogil
 
 
+
+# Function to compute the duality gap
+cdef double duality_gap( # Data
```

Comment:
the code to compute duality_gap for Lasso and E-Net has been moved in separate function.